test/integration/revocation_test.go

```diff
@@ -89,45 +89,70 @@ func getAllCRLs(t *testing.T) map[string][]*x509.RevocationList {
 		// 10 is the number of shards configured in test/config*/crl-updater.json
 		for i := range 10 {
 			crlURL := fmt.Sprintf("%s%d.crl", issuer.CRLURLBase, i+1)
-			resp, err := http.Get(crlURL)
-			if err != nil {
-				t.Fatalf("getting CRL from %s: %s", crlURL, err)
-			}
-			if resp.StatusCode != http.StatusOK {
-				t.Fatalf("fetching %s: status code %d", crlURL, resp.StatusCode)
-			}
-			body, err := io.ReadAll(resp.Body)
-			if err != nil {
-				t.Fatalf("reading CRL from %s: %s", crlURL, err)
-			}
-			resp.Body.Close()
+			list := getCRL(t, crlURL, issuerCert)
 
-			list, err := x509.ParseRevocationList(body)
-			if err != nil {
-				t.Fatalf("parsing CRL from %s: %s (bytes: %x)", crlURL, err, body)
-			}
+			ret[issuerSKID] = append(ret[issuerSKID], list)
+		}
+	}
+	return ret
+}
 
-			err = list.CheckSignatureFrom(issuerCert)
-			if err != nil {
-				t.Errorf("checking CRL signature on %s from %s: %s",
-					crlURL, issuerCert.Subject, err)
-			}
+func getCRL(t *testing.T, crlURL string, issuerCert *x509.Certificate) *x509.RevocationList {
+	resp, err := http.Get(crlURL)
+	if err != nil {
+		t.Fatalf("getting CRL from %s: %s", crlURL, err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		t.Fatalf("fetching %s: status code %d", crlURL, resp.StatusCode)
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("reading CRL from %s: %s", crlURL, err)
+	}
+	resp.Body.Close()
 
-			idpURIs, err := idp.GetIDPURIs(list.Extensions)
-			if err != nil {
-				t.Fatalf("getting IDP URIs: %s", err)
-			}
-			if len(idpURIs) != 1 {
-				t.Errorf("CRL at %s: expected 1 IDP URI, got %s", crlURL, idpURIs)
-			}
-			if idpURIs[0] != crlURL {
-				t.Errorf("fetched CRL from %s, got IDP of %s (should be same)", crlURL, idpURIs[0])
-			}
+	list, err := x509.ParseRevocationList(body)
+	if err != nil {
+		t.Fatalf("parsing CRL from %s: %s (bytes: %x)", crlURL, err, body)
+	}
 
-			ret[issuerSKID] = append(ret[issuerSKID], list)
+	if issuerCert != nil {
```

Comment:
Is there a reason that we're making `issuerCert` optional? I notice that we're not even making it an option to pass into `fetchAndCheckRevoked()` either.

Reply:
In the existing getAllCRLs use case, we're already pulling out the issuer cert file path from the config file. In many of the existing test cases, we didn't already have the issuer cert handy (because we're getting an end entity cert from the ACME API), so it seemed simpler to not pass it.

But looking again, it shouldn't be that hard to extract the issuer from the ACME response, which lets us be consistent about checking the signature. I'll try that.